example-app/src/androidTest/java/backtraceio/backtraceio/ExampleInstrumentedTest.java

```diff
@@ -16,11 +24,46 @@
  */
 @RunWith(AndroidJUnit4.class)
 public class ExampleInstrumentedTest {
+
+    @Rule
+    public ActivityTestRule<MainActivity> mActivityRule =
+            new ActivityTestRule<>(MainActivity.class);
+
+    @Before
+    public void enableMetricsAndBreadcrumbs() {
+        onView(withId(R.id.enableBreadcrumbs)).perform(click());
+    }
+
     @Test
     public void useAppContext() {
         // Context of the app under test.
         Context appContext = InstrumentationRegistry.getTargetContext();
-
         assertEquals("backtraceio.backtraceio", appContext.getPackageName());
     }
+
+    @Test
+    public void dumpWithoutCrash() {
+        onView(withId(R.id.dumpWithoutCrash)).perform(click());
+    }
+
+    @Test(expected = PerformException.class)
+    public void unhandledException() {
+        onView(withId(R.id.unhandledException)).perform(click());
+    }
+
+    @Test
+    public void handledException() {
+        onView(withId(R.id.handledException)).perform(click());
+    }
+
+    @Test
+    public void anr() {
+        onView(withId(R.id.anr)).perform(click());
+    }
+
+    // Will break build, obviously.
+    //@Test
```

Comment:
do we need this comment `//@Test`?

Reply:
Good catch. it’s left there to easily turn on the test. I need to figure out how to run the test (which crashes the app) without crashing the test. Or accept that the test fails but then we need to add a custom annotation that prevent it from running on “normal runs”.